Clarify ServiceAccount use case for private registry image pulls (#56284)
kubernetes/website: content/en/docs/concepts/security/service-accounts.md

* Your Pods need to communicate with an external service. For example, a
  workload Pod requires an identity for a commercially available cloud API,
  and the commercial provider allows configuring a suitable trust relationship.

> * [Providing `imagePullSecrets` for Pods that pull images from private registries](/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account).

Reviewer: Aligning with other items (

```suggestion
* [Your Pods need `imagePullSecrets`to pull images from private registries](/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account).
```
